Automattic/wp-calypso
# Nav redesign - add interface_setting prop to page view analytics

Related to #  surfacing this idea in slack p1709215152834339-slack-C06DN6QQVAQ

## Proposed Changes

* Adds `global_site_view_enabled` prop to page view analytics.
* This takes format from other enhancers such as `enhanceWithSiteType` and `enhanceWithUserIsDevAccount`, and enhances analytics to include an `global_site_view_enabled` prop that will be `0` or `1` depending on whether or not the user is being shown new nav changes for the site.
* Adds this enhancer to the page view analytics.

## Testing Instructions



* Load this branch of calypso
* On a default view site (AT or simple), load a page in calypso. Verify the `global_site_view_enabled` property of the page view event is `0`.
* On a classic view AT site, load a page in calypso. Verify the `global_site_view_enabled` property is `1`.

Tracks vigilante tool helps:

<img width="356" alt="Screenshot 2024-03-01 at 8 52 52 AM" src="https://github.com/Automattic/wp-calypso/assets/28742426/5c4deac3-8f64-4b49-9616-d93e00252aae">


## Pre-merge Checklist
